*** xmlsec/xmlsec-config.in.orig	2006-06-08 13:56:20.000000000 +0200
--- xmlsec/xmlsec-config.in	2006-06-08 14:01:10.000000000 +0200
***************
*** 4,9 ****
--- 4,15 ----
  package="@PACKAGE@"
  exec_prefix="@exec_prefix@"
  exec_prefix_set=no
+ if [ "`ldd /bin/sh | grep lib64`" = "" ]
+ then
+     archlib=lib
+ else
+     archlib=lib64
+ fi
  
  usage()
  {
***************
*** 188,194 ****
  # Assemble all the settings together
  #
  the_flags="$the_flags @XMLSEC_CORE_CFLAGS@ $the_xml_flags $the_xslt_flags $the_crypto_flags"
! the_libs="$the_libs -L${exec_prefix}/lib @XMLSEC_CORE_LIBS@ $the_xmlsec_crypto_lib -lxmlsec1 $the_xml_libs $the_xslt_libs $the_crypto_libs"
  
  if $cflags; then
      all_flags="$the_flags"
--- 194,200 ----
  # Assemble all the settings together
  #
  the_flags="$the_flags @XMLSEC_CORE_CFLAGS@ $the_xml_flags $the_xslt_flags $the_crypto_flags"
! the_libs="$the_libs -L${exec_prefix}/$archlib @XMLSEC_CORE_LIBS@ $the_xmlsec_crypto_lib -lxmlsec1 $the_xml_libs $the_xslt_libs $the_crypto_libs"
  
  if $cflags; then
      all_flags="$the_flags"
